core/src/mindustry/ai/types/DefenderAI.java

```diff
@@ -0,0 +1,34 @@
+package mindustry.ai.types;
+
+import arc.math.*;
+import mindustry.entities.*;
+import mindustry.entities.Units.*;
+import mindustry.entities.units.*;
+import mindustry.gen.*;
+import mindustry.world.meta.*;
+
+public class DefenderAI extends AIController{
+
+    @Override
+    public void updateMovement(){
+        if(target != null){
+            moveTo(target, -1f);
```

Comment:
It should not hug targets. Circle length should be greater than 0 and depend on target hitbox size - octs crashing into eclipses is a bad idea.

Reply:
This still hugs targets, and will crash into Eclipse units.